pkg/crypto/tinkcrypto/primitive/jwe/ecdhes/ecdh_aead_public_key_manager.go

```diff
@@ -0,0 +1,115 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package ecdhes
+
+import (
+	"fmt"
+	"math/big"
+
+	"github.com/golang/protobuf/proto"
+	"github.com/google/tink/go/core/registry"
+	"github.com/google/tink/go/keyset"
+	"github.com/google/tink/go/subtle/hybrid"
+	tinkpb "github.com/google/tink/proto/tink_go_proto"
+
+	sbutleecdhes "github.com/hyperledger/aries-framework-go/pkg/crypto/tinkcrypto/primitive/jwe/subtle/ecdhes"
+	ecdhespb "github.com/hyperledger/aries-framework-go/pkg/crypto/tinkcrypto/primitive/proto/ecdhes_aead_go_proto"
+)
+
+const (
+	ecdhesPublicKeyVersion = 0
+	ecdhesPublicKeyTypeURL = "type.googleapis.com/google.crypto.tink.EcdhesAeadPublicKey"
+)
+
+// common errors
+var errInvalidECDHESPublicKey = fmt.Errorf("ecdhes_public_key_manager: invalid key")
+
+// ecdhesPublicKeyManager is an implementation of KeyManager interface.
+// It generates new ECDHESPublicKey keys and produces new instances of ECDHESPublicKey subtle.
+type ecdhesPublicKeyManager struct{}
+
+// Assert that ecdhesPublicKeyManager implements the KeyManager interface.
+var _ registry.KeyManager = (*ecdhesPublicKeyManager)(nil)
+
+// newECDHESPublicKeyManager creates a new aesGcmKeyManager.
+func newECDHESPublicKeyManager() *ecdhesPublicKeyManager {
+	return new(ecdhesPublicKeyManager)
+}
+
+// Primitive creates an ECDHESPublicKey subtle for the given serialized ECDHESPublicKey proto.
+func (km *ecdhesPublicKeyManager) Primitive(serializedKey []byte) (interface{}, error) {
+	if len(serializedKey) == 0 {
+		return nil, errInvalidECDHESPublicKey
+	}
+
+	key := new(ecdhespb.EcdhesAeadPublicKey)
+
+	err := proto.Unmarshal(serializedKey, key)
+	if err != nil {
+		return nil, errInvalidECDHESPublicKey
+	}
+
+	err = km.validateKey(key)
+	if err != nil {
+		return nil, errInvalidECDHESPublicKey
+	}
+
+	curve, err := hybrid.GetCurve(key.Params.KemParams.CurveType.String())
+	if err != nil {
+		return nil, err
+	}
+
+	pub := hybrid.ECPublicKey{
+		Curve: curve,
+		Point: hybrid.ECPoint{
+			X: new(big.Int).SetBytes(key.X),
+			Y: new(big.Int).SetBytes(key.Y),
+		},
+	}
+
+	rDem, err := newRegisterECDHESEncHelper(key.Params.DemParams.AeadDem)
+	if err != nil {
+		return nil, err
+	}
+
+	ptFormat := key.Params.EcPointFormat.String()
+
+	// TODO add support for multiple 'recipient' keys somehow.
```

Comment:
issue number?

Reply:
I'll add

Reply:
Multiple recipients keys are now supported in this change.